src/modules/LanItems.rb

```diff
@@ -2354,70 +2356,66 @@ def createS390Device(dev_attrs = {})
       case @type
       when "hsi", "qeth"
         @portnumber_param = if Ops.greater_than(Builtins.size(@qeth_portnumber), 0)
-          Builtins.sformat("-n %1", @qeth_portnumber)
+          Builtins.sformat("-n %1", @qeth_portnumber.to_s.shellescape)
         else
           ""
         end
         @portname_param = if Ops.greater_than(Builtins.size(@qeth_portname), 0)
-          Builtins.sformat("-p %1", @qeth_portname)
+          Builtins.sformat("-p %1", @qeth_portname.shellescape)
         else
           ""
         end
         @options_param = if Ops.greater_than(Builtins.size(@qeth_options), 0)
-          Builtins.sformat("-o %1", @qeth_options)
+          Builtins.sformat("-o %1", @qeth_options.shellescape)
         else
           ""
         end
         command1 = Builtins.sformat(
-          "qeth_configure %1 %2 %3 %4 %5 1",
+          "/sbin/qeth_configure %1 %2 %3 %4 %5 1",
           @options_param,
           @qeth_layer2 ? "-l" : "",
           @portname_param,
           @portnumber_param,
           @qeth_chanids
         )
```

Comment:
couldn't all that escaping above replaced simply by ```command1 = <formating>.shellescape``` ?

Reply:
similarly for the ```command2```

Reply:
i mean somewhere after the ```case``` before execution.

Reply:
I worry not, because we no longer know what is param and what is space in original variable.

Reply:
ok